src/utils/db/db.ts

```diff
@@ -698,52 +706,73 @@ export class DB {
         }
     }
 
-    public async createPluginLogEntry(
-        pluginConfig: PluginConfig,
-        source: PluginLogEntrySource,
-        type: PluginLogEntryType,
-        message: string,
-        instanceId: UUID,
-        timestamp: string = new Date().toISOString()
-    ): Promise<PluginLogEntry> {
-        const entry: PluginLogEntry = {
-            id: new UUIDT().toString(),
-            team_id: pluginConfig.team_id,
-            plugin_id: pluginConfig.plugin_id,
-            plugin_config_id: pluginConfig.id,
-            timestamp: timestamp.replace('T', ' ').replace('Z', ''),
-            source,
-            type,
-            message,
-            instance_id: instanceId.toString(),
+    public async createPluginLogEntries(pluginConfig: PluginConfig, entries: LogEntryPayload[]): Promise<void> {
+        const parsedEntries: PluginLogEntry[] = entries.map((entry) => {
+            const { source, message, type, timestamp, instanceId } = entry
+            this.statsd?.increment(`logs.entries_created`, {
+                source,
+                team_id: pluginConfig.team_id.toString(),
+                plugin_id: pluginConfig.plugin_id.toString(),
+            })
+            return {
+                id: new UUIDT().toString(),
+                team_id: pluginConfig.team_id,
+                plugin_id: pluginConfig.plugin_id,
+                plugin_config_id: pluginConfig.id,
+                timestamp: (timestamp || new Date().toISOString()).replace('T', ' ').replace('Z', ''),
+                source,
+                type,
+                message,
+                instance_id: instanceId.toString(),
+            }
+        })
+
+        if (this.kafkaProducer) {
+            for (const parsedEntry of parsedEntries) {
+                try {
+                    await this.kafkaProducer.queueMessage({
+                        topic: KAFKA_PLUGIN_LOG_ENTRIES,
+                        messages: [{ key: parsedEntry.id, value: Buffer.from(JSON.stringify(parsedEntry)) }],
+                    })
+                } catch (e) {
+                    captureException(e)
+                    console.error(parsedEntry)
+                    console.error(e)
+                }
+            }
+            return
```

Comment:
@mariusandra this should prevent the insert into postgres

Reply:
Oh god, that's embarrassing :D. 

In my defence I have a strong cold, but still... :D